data/src/test/java/org/apache/iceberg/data/TestLocalScan.java

```diff
@@ -225,12 +222,13 @@ public void createTables() throws IOException {
     sharedTable.newAppend().appendFile(file1).appendFile(file2).appendFile(file3).commit();
   }
 
-  @Test
+  @TestTemplate
   public void testRandomData() throws IOException {
     List<Record> expected = RandomGenericData.generate(SCHEMA, 1000, 435691832918L);
 
-    File location = temp.newFolder(format.name());
-    Assert.assertTrue(location.delete());
+    File location = File.createTempFile("junit", null, tempDir);
```

Comment:
seems like this should be also creating a temp directory rather than a temp file

Reply:
yep, done